src/auth/auth.ts

```diff
@@ -1107,7 +1109,8 @@ class AuthService {
 
 		await this.sendEmailVerificationToken(user, network);
 		// send undo token in background
-		sendUndoEmailChangeEmail(user, newUndoEmailChangeToken, network);
+		if(shouldSendUndoEmailChangeEmail)
+			sendUndoEmailChangeEmail(user, newUndoEmailChangeToken, network);

```

Comment:
we shouldn't generate newUndoEmailChangeToken if shouldSendUndoEmailChangeEmail is false